src/libs/actions/Report.js

```diff
@@ -1159,26 +1169,32 @@ Onyx.connect({
  */
 function deleteReportComment(reportID, reportAction) {
     const sequenceNumber = reportAction.sequenceNumber;
-
-    // We are not updating the message content here so the message can re-appear as strike-throughed
-    // if the user goes offline. The API will update the message content to empty strings on success.
+    const deletedMessage = [{
+        type: 'COMMENT',
+        html: '',
+        text: '',
+        isEdited: true,
+    }];
     const optimisticReportActions = {
         [sequenceNumber]: {
             pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE,
+            previousMessage: reportAction.message,
```

Comment:
Mostly the change to actually delete the actions optimistically instead of leave them in (as described [here](https://github.com/Expensify/App/blob/b4b12addac16a3ef3753273e25c3e58379715a76/src/libs/actions/Report.js#L1163-L1164)) makes more sense to my brain. 

I didn't love that when we are optimistically deleting an action it wasn't _actually_ reflecting the "optimistic response" and instead we would leave it untouched and have the `OfflineWithFeedback` component hide the children based on the `pendingAction`. 

Mainly to the rest of the app the action looks like it is "not deleted", but I think for most purposes it should appear to as "deleted" e.g. when calculating what the last message of a report or `lastReadSequenceNumber` should be.

So, to clarify the statement in the description... you are right it seems like it works on `main`, but I noticed this wasn't working locally after I made the decision to unset the message optimistically on delete instead of leaving it in place.

Since you worked on this feature before, do you agree with the changes I've made or see any issues that I might not have caught?

Reply:
Totally agree with the changes and the code looks cleaner :D